distribution/sc-proxy-dex/src/proxy_common.rs

```diff
@@ -31,6 +35,76 @@ pub trait ProxyCommonModule {
         Ok(())
     }
 
+    #[payable("*")]
+    #[endpoint]
+    fn acceptPay(
+        &self,
+        #[payment_token] token_id: TokenIdentifier,
+        #[payment] amount: Self::BigUint,
+    ) {
+        let tx_hash = self.blockchain().get_tx_hash();
+
+        if !self.last_tx_hash().is_empty() {
+            let last_tx_hash = self.last_tx_hash().get();
+            if tx_hash != last_tx_hash {
+                self.last_tx_hash().set(&tx_hash);
+                self.last_tx_accepted_funds().clear();
+            }
+        } else {
+            self.last_tx_hash().set(&tx_hash);
+        }
+
+        let token_nonce = self.call_value().esdt_token_nonce();
+        self.last_tx_accepted_funds()
+            .insert((token_id, token_nonce), amount);
+    }
+
+    fn validate_received_funds_on_current_tx(
```

Comment:
I think you need one more verification - if I want to be picky. To verify that you did not get some funds which you did not want ? like you wanted exactly this token - and only that. none others. If there are 2 token IDs and you wanted only one - the tx should fail.

Reply:
done